lib/lz4.c

```diff
@@ -414,6 +414,7 @@ typedef enum { full = 0, partial = 1 } earlyEnd_directive;
 *  Local Utils
 **************************************/
 int LZ4_versionNumber (void) { return LZ4_VERSION_NUMBER; }
+const char* LZ4_versionString (void) { return LZ4_VERSION_STRING; }
```

Comment:
Yes it will. There's nothing special about `printf`, adjacent string literals are always concatenated in C.